lib/spdy-transport/protocol/http2/parser.js

```diff
@@ -345,6 +371,11 @@ Parser.prototype.onRSTFrame = function onRSTFrame(header, body, callback) {
     return callback(this.error(constants.error.PROTOCOL_ERROR,
                                'Invalid stream id for RST_STREAM'));
   }
+  
+  if (this._lastHeaderBlock === null && this.pendingHeader === null) {
```

Comment:
I don't think that this is correct. It looks like this will just hide the issue, without actually fixing it.

The proper way should be to handle this on connection level, where the information about idle-ness of the stream should be available.